app/src/components/main/colorPicker/colorPicker.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import * as ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
+import { CustomPicker, CirclePicker, ChromePicker } from 'react-color';
+import { Manager, Reference, Popper } from 'react-popper';
+
+import styles from './colorPicker.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  pickSwatch: {
+    id: 'ColorPicker.pickSwatch',
+    defaultMessage: 'Pick a swatch',
+  },
+  selectColor: {
+    id: 'ColorPicker.selectColor',
+    defaultMessage: 'Select your color',
+  },
+});
+
+@injectIntl
+class ColorPickerComponent extends Component {
+  static propTypes = {
+    hex: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  state = {
+    opened: false,
+  };
+
+  toggleColorPicker = () => {
+    this.setState({ opened: !this.state.opened });
+  };
+
+  COLORS = [
+    '#f48fb1',
+    '#cc92d6',
+    '#b39ddb',
+    '#81d4fa',
+    '#8de1ec',
+    '#80cbc4',
+    '#c5e1a5',
+    '#e6ee9c',
+    '#ffcc80',
+    '#ffab91',
+    '#f50057',
+    '#d500f9',
+    '#651fff',
+    '#00b0ff',
+    '#00e5ff',
+    '#1de9b6',
+    '#76ff03',
+    '#c6ff00',
+    '#ffc400',
+    '#ff3d00',
+    '#ad1457',
+    '#6a1b9a',
+    '#4527a0',
+    '#0277bd',
+    '#00838f',
+    '#00695c',
+    '#558b2f',
+    '#9e9d24',
+    '#ff8f00',
+    '#d84315',
+  ];
+
+  render() {
+    const { hex, onChange, intl } = this.props;
+    const { opened } = this.state;
+
+    return (
+      <Manager>
+        <Reference>
+          {({ ref }) => (
+            <button
+              type="button"
+              ref={ref}
+              className={cx('control', { 'control-opened': opened })}
+              onClick={this.toggleColorPicker}
+            >
+              <span className={cx('color-marker')} style={{ backgroundColor: hex }} />
+            </button>
+          )}
+        </Reference>
+        {opened &&
+          ReactDOM.createPortal(
+            <Popper placement="bottom-start">
+              {({ ref, style, placement }) => (
+                <div
+                  className={cx('color-picker')}
+                  ref={ref}
+                  style={style}
+                  data-placement={placement}
+                  onClick={(e) => e.stopPropagation()}
+                >
+                  <div className={cx('title')}>{intl.formatMessage(messages.pickSwatch)}:</div>
+                  <CirclePicker
+                    color={hex}
+                    onChange={onChange}
+                    colors={this.COLORS}
```

Comment:
May be it would be better to setup color picker settings as constants, which would be better to put in a separate file.

Reply:
done